lib/http1.c

```diff
@@ -353,11 +354,14 @@ static ssize_t fixup_request(struct st_h2o_http1_conn_t *conn, struct phr_header
     entity_header_index =
         init_headers(&conn->req.pool, &conn->req.headers, headers, num_headers, &connection, &host, &upgrade, expect);
 
+    if (h2o_url_parse(conn->req.input.path.base, conn->req.input.path.len, &url) == 0)
```

Comment:
This is a hot path and I would prefer having a fast check to see if we need to call the `h2o_url_parse` function.

Reply:
Good idea, addressed in ce1855c